addons/delivery/models/product_product.py

```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def _write_sale_order_line(self, sale_order, sale_order_line):
+        sale_order.write({'recompute_delivery_price': True})
+        super()._write_sale_order_line(sale_order, sale_order_line)
```

Comment:
Dead code

Reply:
Edit: should be handled by calling `onchange_order_line` in an override of the controller update route